src/pykeen/training/training_loop.py

```diff
@@ -650,14 +653,20 @@ def _forward_pass(self, batch, start, stop, current_batch_size, label_smoothing,
             loss *= (this_sub_batch_size / current_batch_size)
 
         # backward pass
-        loss.backward()
+        self._loss_backward(loss)
         current_epoch_loss = loss.item()
 
         self.model.post_forward_pass()
         # TODO why not call torch.cuda.empty_cache()? or call self._free_graph_and_cache()?
 
         return current_epoch_loss
 
+    def _prepare_training(self, data):
```

Comment:
just from reading this file, this looks magic. Can we add a small docstring here, that this is a hook / interaction point where we can prepare the data loader (?) for training?